releng/feature-announce/feature-announce.go

```diff
@@ -0,0 +1,318 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright the KubeVirt Authors.
+ */
+
+package main
+
+import (
+	"context"
+	_ "embed"
+	"flag"
+	"fmt"
+	"github.com/google/go-github/github"
+	log "github.com/sirupsen/logrus"
+	"golang.org/x/oauth2"
+	"k8s.io/test-infra/prow/config/secret"
+	"os"
+	"os/exec"
+	"regexp"
+	"strconv"
+	"strings"
+	"text/template"
+)
+
+type options struct {
+	dryRun          bool
+	logLevel        int
+	githubTokenFile string
+	org             string
+	repo            string
+	repositoryPath  string
+	outputFile      string
+}
+
+//go:embed "upcoming-changes.gomd"
+var upcomingChangesTemplate string
+
+func (o *options) Validate() error {
+	if o.org == "" {
+		return fmt.Errorf("org is required")
+	}
+	if o.repo == "" {
+		return fmt.Errorf("repo is required")
+	}
+	if o.githubTokenFile == "" {
+		return fmt.Errorf("github-token-file is required")
+	}
+	if o.repositoryPath == "" {
+		return fmt.Errorf("path-to-repository is required")
+	}
+	if o.outputFile == "" {
+		temp, err := os.CreateTemp("", "upcoming-changes-*.md")
+		if err != nil {
+			return err
+		}
+		o.outputFile = temp.Name()
+	}
+	return nil
+}
+
+func gatherOptions() options {
+	o := options{}
+	flag.IntVar(&o.logLevel, "V", int(log.WarnLevel), "log level from logrus")
```

Comment:
Could we use a flag name like `verbose` here? I always associate `v` with version - i also noticed the rest of the flags are not single characters.

Reply:
Hm, I was trying to do something in the lines of `glog`, where they use `v` IIRC, and I faintly remember some other go based tools where I had seen that. 

I'd not call it `verbose` but rather `log-level` then. WDYT?

Reply:
`log-level` sounds great to me.

Reply:
I don't see this updated?

Reply:
I haven't yet done it, sorry for being so slow.

Reply:
Updated